src/cmd-import

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/python3
+
+import argparse
+import json
+import os
+import subprocess
+import tempfile
+import shutil
+import sys
+from stat import (
+    S_IREAD,
+    S_IRGRP,
+    S_IROTH)
+from cosalib.builds import Builds
+from cosalib.cmdlib import (
+    import_ostree_commit,
+    get_basearch,
+    sha256sum_file)
+
+VALID_NAMES = ["fedora-coreos", "rhcos", "scos"]
+
+
+def main():
+    args = parse_args()
+    with tempfile.TemporaryDirectory(prefix='cosa-import-', dir='tmp') as tmpd:
+        populate_build_dir(args, tmpd)
+        # XXX: for debugging until we're ready
+        print("Temporary directory is:", tmpd)
+        sys.exit()
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(prog='cosa import')
+    parser.add_argument("--rechunk", action='store_true', help="rechunk image")
+    parser.add_argument("srcimg", metavar='IMAGE',
+                        help="image to import (containers-transports(5) format)")
+    parser.add_argument("--name", help="name to be set for the imported image. (eg: fedora-coreos)")
+    return parser.parse_args()
+
+def extract_name(args):
+    if args.name:
+        if args.name in VALID_NAMES:
+            return args.name
+        raise Exception(f'{args.name} is not a valid name')
+
+    for name in VALID_NAMES:
+        if name in args.srcimg:
+            return name
+
+    raise Exception("unable to guess name for the image. Use --name option to set a name")
+
+
+def prepare_build(builds, buildid):
+    pwd = os.getcwd()
+    builddir = builds.get_build_dir(buildid)
+    buildmeta = builds.get_build_meta(buildid)
+    import_ostree_commit(pwd, builddir, buildmeta, extract_json=0)
+
+
+def populate_build_dir(args, tmpd):
+    name = extract_name(args)
+    target_ociarchive = os.path.join(tmpd, "out.ociarchive")
+    import_oci_archive(args, target_ociarchive)
+    metadata = inspect_oci_archive(target_ociarchive)
+    buildid = metadata['Labels']['org.opencontainers.image.version']
+    arch = get_basearch()
+
+    os.makedirs(f'builds/{buildid}/{arch}/', exist_ok=True)
+
+    manifest_metadata = generate_manifest_json(target_ociarchive, name, buildid, arch)
+
+    meta_json = generate_meta_json(target_ociarchive, metadata, manifest_metadata, name)
+
+    archive_path = meta_json['images']['ostree']['path']
+
+    # Move ociarchive to build dir
+    shutil.move(target_ociarchive, f'builds/{buildid}/{arch}/{archive_path}')
+
+    # Symlink build to latest
+    if os.path.exists('builds/latest'):
+        os.remove('builds/latest')
+    os.symlink(f'{buildid}', 'builds/latest', target_is_directory=True)
+
+    builds = Builds()
+    update_builds_json(builds, buildid, arch)
+    prepare_build(builds, buildid)
```

Comment:
Let's do this instead in a separate function called from the top-level. The idea is that `prepare_build()` prepares the temporary build dir, and then a new e.g. `finalize_build()` moves it into the production location.